### PR TITLE
add support for Shields theming

### PR DESCRIPTION
**Pull request by cezaraugusto**
fix https://github.com/brave/brave-browser/issues/3870

Test Plan:

1. Go to chrome://settings/appearance
2. Switch theme
3. Shields should match whether dark/light

### CODE REVIEW

File: components/brave_extension/extension/brave_extension/background/reducers/shieldsPanelReducer.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ const updateShieldsIconImage = (state: State) => {
 const updateShieldsIcon = (state: State) => {
   updateShieldsIconBadgeText(state)
   updateShieldsIconImage(state)
+  shieldsPanelState.updateThemeType(state)
```
Comment (petemill) [inline components/brave_extension/extension/brave_extension/background/reducers/shieldsPanelReducer.ts]: Hey @cezaraugusto just wondering why we're calling `updateThemeType` inside `updateShieldsIcon`. I'm not sure I see the connection between the shields icon being affected by state of the page and shield-blocking-state and the brave theme being affected by user choice.

File: components/brave_extension/extension/brave_extension/types/state/shieldsPannelState.ts [hunk 2]
```diff
@@ -43,6 +44,7 @@ export interface State {
   currentWindowId: number
   tabs: Tabs
   windows: Windows
+  theme: ThemeType
```
Comment (petemill) [inline components/brave_extension/extension/brave_extension/types/state/shieldsPannelState.ts]: Should we really add a `theme` prop to specific state for each UI we have? I'm not sure and think it's adding noise and complication. The data reducers don't depend on this property at the moment, and if we do want it on state, we can use a shared store/reducer that can be imported for any page which wants it, e.g. UIThemeStore, combined via `combineReducers`.

File: components/brave_extension/extension/brave_extension/braveShieldsPanel.tsx [hunk 3]
```diff
@@ -18,13 +20,14 @@ chrome.storage.local.get('state', (obj) => {
   const store: any = new Store({
     portName: 'BRAVE'
   })
-
   store.ready()
     .then(() => {
       const mountNode: HTMLElement | null = document.querySelector('#root')
+      const theme: ThemeType = store.state.shieldsPanel.theme
```
Comment (petemill) [inline components/brave_extension/extension/brave_extension/braveShieldsPanel.tsx]: Correct me if I'm wrong here, but reading from state here will never result in an update when state changes, so it's not serving a useful data-flow purpose?

**Comment by petemill** [comment]
Here's a POC that handles `chrome.braveTheme.onBraveThemeTypeChanged` with a new Component, `BraveCoreThemeProvider`: https://github.com/brave/brave-core/compare/try-shields-theme...try-shields-theme-onchange.

However, @simonhong, I'm not seeing that function get called when the OS theme changes, which is the only case that the shields will handle. If you click in the browser window to change the theme, then the shields panel will close, but if you click in the macOS Settings window to change the OS theme, the panel will stay open and be the incorrect theme:

**Comment by cezaraugusto** [comment]
after convo with @petemill we decided that https://github.com/brave/brave-core/pull/2084 is preferable for theming as it's simpler and future-friendly for theming other features so closing this.

**Pull request closed without merging**